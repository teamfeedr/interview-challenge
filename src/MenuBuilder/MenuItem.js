import React from 'react';
import DieteriesList from './DietariesList';

const MenuItem = ({ id, name, dietaries, remove = () => {} }) => (
  <li className="item">
    <h2>{name}</h2>
    <DieteriesList dietaries={dietaries} />
    {
      <button className="remove-item" onClick={() => remove()}>
        x
      </button>
    }
  </li>
);

export default MenuItem;
