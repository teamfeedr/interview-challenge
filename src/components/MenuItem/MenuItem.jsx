import React from 'react';
import PropTypes from 'prop-types';

import DietaryIcon from '../DietaryIcon/DietaryIcon';

const getDietaryIconElements = ({ dietaries }) =>
    dietaries.map((diet) => (<DietaryIcon diet={diet} key={diet} />));

const MenuItem = ({ handleOnSelect, handleOnDeselect, dietaries, name, id }) => {
    const dietaryIcons = getDietaryIconElements({ dietaries });
    const showRemoveIcon = handleOnDeselect;

    return (
        <li key={id} className="item" onClick={handleOnSelect}>
            <h2>{name}</h2>
            <p>
                {dietaryIcons}
            </p>
            {showRemoveIcon && (
                <button className="remove-item" onClick={handleOnDeselect}>x</button>
            )}
        </li>
    )
}

MenuItem.propTypes = {
    dietaries: PropTypes.arrayOf(PropTypes.string).isRequired,
    name: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    handleOnDeselect: PropTypes.func,
    handleOnSelect: PropTypes.func
}

MenuItem.defaultProps = {
    handleOnDeselect: undefined,
    handleOnSelect: undefined
}

export default MenuItem;