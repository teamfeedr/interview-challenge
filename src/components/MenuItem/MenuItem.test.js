import React from "react";
import { shallow } from "enzyme";

import MenuItem from "./MenuItem";

describe("the MenuItem component", () => {
  it("renders with a list of dietaries", () => {
    const rendered = shallow(<MenuItem dietaries={["v", "gf"]} name={"testName"} id={1} />);

    expect(rendered).toMatchSnapshot();
  });

  it("renders with a remove button", () => {
    const rendered = shallow(<MenuItem dietaries={['v', 'gf']} name={'testItem'} id={1} handleOnDeselect={() => {}}/>);

    expect(rendered).toMatchSnapshot();
  });
});
