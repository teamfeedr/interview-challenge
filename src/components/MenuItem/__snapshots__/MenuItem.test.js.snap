// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the MenuItem component renders with a list of dietaries 1`] = `
<li
  className="item"
  key="1"
>
  <h2>
    testName
  </h2>
  <p>
    <DietaryIcon
      diet="v"
      key="v"
    />
    <DietaryIcon
      diet="gf"
      key="gf"
    />
  </p>
</li>
`;

exports[`the MenuItem component renders with a remove button 1`] = `
<li
  className="item"
  key="1"
>
  <h2>
    testItem
  </h2>
  <p>
    <DietaryIcon
      diet="v"
      key="v"
    />
    <DietaryIcon
      diet="gf"
      key="gf"
    />
  </p>
  <button
    className="remove-item"
    onClick={[Function]}
  >
    x
  </button>
</li>
`;
